src/esrs/aggregate.rs

```diff
@@ -110,80 +112,40 @@ pub trait AggregateManager: Identifier {
         }
     }
 
-    /// Persists an event into the event store - recording it in the aggregate instance's history.
-    async fn persist(
+    /// Transactional persists events in store - recording it in the aggregate instance's history.
+    /// The store will also project the events. If an error occurs whilst persisting the events,
+    /// the whole transaction is rolled back and the error is returned.
+    ///
+    /// The policies associated to the store are run here. A failure at this point will be silently
+    /// ignored, and the new state returned successfully anyway.
```

Comment:
These two lines do not represent the actual code. We return early with an error if any policy fails.